go/consensus/tendermint/tendermint.go

```diff
@@ -1173,17 +1179,37 @@ func (t *tendermintService) syncWorker() {
 		case <-t.node.Quit():
 			return
 		case <-time.After(1 * time.Second):
-			isSyncing, err := checkSyncFn()
+			isFastSyncing, err := checkSyncFn()
 			if err != nil {
 				t.Logger.Error("Failed to poll FastSync",
 					"err", err,
 				)
 				return
 			}
-			if !isSyncing {
+			if !isFastSyncing {
 				t.Logger.Info("Tendermint Node finished fast-sync")
-				close(t.syncedCh)
-				return
+
+				// Check latest block time.
+				tmBlock, err := t.GetTendermintBlock(t.ctx, consensusAPI.HeightLatest)
+				if err != nil {
+					t.Logger.Error("Failed to get tendermint block",
+						"err", err,
+					)
+					return
+				}
+
+				now := time.Now()
+				// No committed blocks or latest block within threshold.
+				if tmBlock == nil || now.Sub(tmBlock.Header.Time) < syncWorkerLastBlockTimeDiffThreshold {
+					close(t.syncedCh)
```

Comment:
Consider logging something when this happens.

Reply:
done, also added a test